app/src/fdroid/java/com/google/android/stardroid/activities/util/GooglePlayServicesChecker.java

```diff
@@ -0,0 +1,106 @@
+package com.google.android.stardroid.activities.util;
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.FragmentManager;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.support.v4.app.ActivityCompat;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.android.stardroid.R;
+import com.google.android.stardroid.activities.DynamicStarMapActivity;
+import com.google.android.stardroid.activities.dialogs.LocationPermissionRationaleFragment;
+import com.google.android.stardroid.control.LocationController;
+import com.google.android.stardroid.util.MiscUtil;
+
+import javax.inject.Inject;
+
+/**
+ * Created by johntaylor on 4/2/16.
+ */
+public class GooglePlayServicesChecker implements LocationPermissionRationaleFragment.Callback {
+  private static final String TAG = MiscUtil.getTag(GooglePlayServicesChecker.class);
+  private final Activity parent;
+  private final SharedPreferences preferences;
+  private final LocationPermissionRationaleFragment rationaleDialog;
+  private final FragmentManager fragmentManager;
+
+  @Inject
+  GooglePlayServicesChecker(Activity parent, SharedPreferences preferences,
+                            LocationPermissionRationaleFragment rationaleDialog,
+                            FragmentManager fragmentManager) {
+    this.parent = parent;
+    this.preferences = preferences;
+    this.rationaleDialog = rationaleDialog;
+    this.fragmentManager = fragmentManager;
+    rationaleDialog.setCallback(this);
+  }
+
+  /**
+   * Checks whether play services is available and up to date and prompts the user
+   * if necessary.
+   * <p/>
+   * Note that at present we only need it for location services so if the user is setting
+   * their location manually we don't do the check.
+   */
+  public void maybeCheckForGooglePlayServices() {
+    Log.d(TAG, "Google Play Services check");
+    if (preferences.getBoolean(LocationController.NO_AUTO_LOCATE, false)) {
+      Log.d(TAG, "Auto location disabled - not checking for GMS");
+      return;
+    }
+    checkLocationServicesEnabled();
+  }
+
+  private void checkLocationServicesEnabled() {
```

Comment:
There is a good amount of duplicate code here that also exists in the `gms` variant file.  I think the only difference is the method `maybeCheckForGooglePlayServices()`?  If so, this should be refactored so that there is an abstract class in the main folder with extended classes that implement this method in the individual build flavors.